desktop/apps/article/components/App.js

```diff
@@ -36,11 +38,29 @@ export default class App extends React.Component {
   }
 
   renderArticle = () => {
-    const { article, isMobile, isSuper, subscribed } = this.props
+    let { article } = this.props
+    const { isMobile, isSuper, subscribed } = this.props
     const articleMarginTop = article.layout === 'standard' ? '100px' : '0px'
     const navHeight = isSuper ? '0px' : NAVHEIGHT
     const headerHeight = `calc(100vh - ${navHeight})`
 
+    /**
+     * FIXME:
+     * Patch missing canvas cover images with display cover images. Needed
+     * until support is added in Positron.
+     */
+    const cover_image_url = get(article, 'display.panel.cover_image_url', false)
+
+    if (cover_image_url) {
+      article = updeep({
+        display: {
+          canvas: {
+            cover_image_url
+          }
+        }
+      }, article)
```

Comment:
@kanaabe, @eessex - We use [updeep](https://github.com/substantial/updeep) all the time in Redux code and is a great tool for updating nested data structures (amongst other things).

Reply:
(Just don't forget to pass in the object you want to update as the second param! I've killed hours in the past only to realize that i was missing that)

Reply:
Oo stoked to try this. This could work well in Positron's api as well, where we use `cloneDeep` to do updates.